### PR TITLE
Expose configuration metadata groups

### PR DESCRIPTION
**Pull request by ericbottard**
This is needed for spring-cloud/spring-cloud-dataflow#187 and better
TAB completion

This resolves spring-cloud/spring-cloud-dataflow#187.

### CODE REVIEW

File: spring-cloud-stream-configuration-metadata/src/main/java/org/springframework/cloud/stream/configuration/metadata/ModuleConfigurationMetadataResolver.java [hunk 1]
```diff
@@ -45,6 +46,46 @@ public ModuleConfigurationMetadataResolver() {
...
 
 	/**
+	 * Return metadata about configuration properties (as groups) that are documented via
+	 * <a href="http://docs.spring.io/spring-boot/docs/current/reference/html/configuration-metadata.html">
+	 * Spring Boot configuration metadata</a> and visible in a module.
+	 *
+	 * @param module a Spring Cloud Stream module; typically a Boot uberjar,
+	 * but directories are supported as well
+	 */
+	public List<ConfigurationMetadataGroup> listPropertyGroups(Resource module) {
```
Comment (ilayaperumalg) [inline spring-cloud-stream-configuration-metadata/src/main/java/org/springframework/cloud/stream/configuration/metadata/ModuleConfigurationMetadataResolver.java]: It appears there are some common steps in the `methods` defined in this class. Can we refactor those?

Comment (ilayaperumalg) [inline spring-cloud-stream-configuration-metadata/src/main/java/org/springframework/cloud/stream/configuration/metadata/ModuleConfigurationMetadataResolver.java]: @ericbottard Could you help me understand how are the methods in this class invoked?

Comment (ericbottard) [inline spring-cloud-stream-configuration-metadata/src/main/java/org/springframework/cloud/stream/configuration/metadata/ModuleConfigurationMetadataResolver.java]: See https://github.com/spring-cloud/spring-cloud-dataflow/pull/199

**Comment by ghillert** [comment]
LGTM - merged with commit 81b04de2227593cc29c75fb95cff7bd45a708bbb

**Pull request closed without merging**